misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql

```diff
@@ -172,21 +192,28 @@
             {% set managed = results[0] %}
             {% set size = results[1] %}
             {% set replication_factor = results[2] %}
+            {% set schedule_type = results[5] %}
+            {% set refresh_rehydration_time_estimate = results[6] %}
 
             {% if not managed %}
                 {{ exceptions.raise_compiler_error("Production cluster " ~ cluster ~ " is not managed") }}
             {% endif %}
 
             {% set create_cluster %}
                 CREATE CLUSTER {{ deploy_cluster }} (
-                    SIZE = {{ dbt.string_literal(size) }},
-                    REPLICATION FACTOR = {{ replication_factor }}
+                    SIZE = {{ dbt.string_literal(size) }}
+                    {% if schedule_type == 'manual' %}
+                        , REPLICATION FACTOR = {{ replication_factor }}
+                    {% elif schedule_type == 'on-refresh' and refresh_rehydration_time_estimate %}
+                        , SCHEDULE = ON REFRESH (REHYDRATION TIME ESTIMATE = {{ dbt.string_literal(refresh_rehydration_time_estimate) }})
+                    {% endif %}
```

Comment:
The additional check here is to make sure that the replication factor is only applied if the schedule type is manual:
```
ERROR:  REPLICATION FACTOR cannot be given together with any SCHEDULE other than MANUAL
```

Reply:
That's right; you can't manually tune the `REPLICATION FACTOR` of a scheduled cluster.